grails-app/services/org/pih/warehouse/inventory/InventoryItemDataService.groovy

```diff
@@ -0,0 +1,15 @@
+package org.pih.warehouse.inventory
+
+import grails.gorm.services.Join
+import grails.gorm.services.Service
+
+@Service(InventoryItem)
+interface InventoryItemDataService {
+
+    InventoryItem save(InventoryItem inventoryItem);
+
+    @Join("product")
+    InventoryItem get(String id);
```

Comment:
I don't like this, because `get` is somewhat "basic" operation and we don't always want to fetch the product. It is a similar case if you made the `product` as fetch type eager by default.

Don't get me wrong - it's a good solution, but the method should be named `getWithProduct`/`findWithProduct` or something like this (check what is working): a resource that might help: https://gorm.grails.org/latest/hibernate/manual/index.html#dataServices

But I would leave `get` without any join fetches.